### PR TITLE
MM-58847 Sanitize User

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary

This PR adds fields to the `SanitizeInput` method. It also adds a test for setting the object's `ID` on an insert. 
It also updates the documentation In a couple cases to include additional fields the code already allows. 

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-58847

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by mgdelacroix — APPROVED** [review]

**Review by hanzei — APPROVED** [review]
LGTM :+1:

File: server/channels/api4/user.go [hunk 1]
```diff
@@ -1233,6 +1233,7 @@ func updateUser(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	user.SanitizeInput(c.IsSystemAdmin())
```
[the remark this answers is not in the document]
Comment (sbishel) [reply server/channels/api4/user.go]: Outdated code - `SantiizeInput` is not called on `updateUser`

**Review by lindy65 — APPROVED** [review]
Thanks @sbishel - QA deferring to be tested post merge

**Review by esarafianou — APPROVED** [review]

**Pull request merged**